app/lib/katello/agent/client_message_handler.rb

```diff
@@ -34,9 +34,17 @@ def handle
       private
 
       def handle_dynflow_event
-        task_exists = ForemanTasks::Task.exists?(external_id: @dispatch_history.dynflow_execution_plan_id, result: 'pending')
-        unless task_exists
-          logger.warn("Couldn't find pending task with external_id=#{@dispatch_history.dynflow_execution_plan_id} dispatch_history_id=#{@dispatch_history.id}")
+        return unless accepted? || result
```

Comment:
No real behavior change here, I just improved the logging a bit to it's easier to follow when debugging this thing